.github/workflows/check-formatting.yml

```diff
@@ -0,0 +1,21 @@
+name: Check formatting
+on: workflow_call
+
+jobs:
+  pre-commit:
+    name: Run pre-commit hooks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the latest code (shallow clone)
+        uses: actions/checkout@v4
+      - name: Run Pre-Commit checks
+        run: cp .github/check-format.sh /tmp && /tmp/check-format.sh
+        id: pre-commit
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Upload code formatting patch
+        if: ${{ failure() && steps.pre-commit.outcome == 'failure' }}
```

Comment:
why does it need two failure?

Reply:
because I don't want to trigger this if any other step failed, e.g. the git checkout step. But I guess it's not necessary, since the uploading would just also fail. 
I will use only one failure.

Reply:
I think keeping the second one should be enough? or the second one requires `failure()` to properly catch

Reply:
The first one is necessary, see: https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions. If `failure()` is not used, the `if` will implicitly use `success()`.

Reply:
I see. Thanks for explanation. Keeping both looks good to me